systemtest/src/test/java/io/strimzi/systemtest/kafka/KafkaST.java

```diff
@@ -1081,212 +1129,193 @@ void testLabelModificationDoesNotBreakCluster(ExtensionContext extensionContext)
 
-        assertThat("Label exists in kafka pods", labelValues[0].equals(labels.get(labelKeys[0])));
-        assertThat("Label exists in kafka pods", labelValues[1].equals(labels.get(labelKeys[1])));
-        assertThat("Label exists in kafka pods", labelValues[2].equals(labels.get(labelKeys[2])));
+        for (ConfigMap configMap : configMaps) {
+            LOGGER.info("Getting labels from {} config map", configMap.getMetadata().getName());
+            verifyAppLabelsForSecretsAndConfigMaps(configMap.getMetadata().getLabels());
+        }
 
-        LOGGER.info("Removing labels: {} -> {}, {} -> {}, {} -> {}", labelKeys[0], labels.get(labelKeys[0]),
-            labelKeys[1], labels.get(labelKeys[1]), labelKeys[2], labels.get(labelKeys[2]));
+        LOGGER.info("---> PVC (both labels and annotation) <---");
 
-        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), resource -> {
-            resource.getMetadata().getLabels().remove(labelKeys[0]);
-            resource.getMetadata().getLabels().remove(labelKeys[1]);
-            resource.getMetadata().getLabels().remove(labelKeys[2]);
-        }, testStorage.getNamespaceName());
+        List<PersistentVolumeClaim> pvcs = kubeClient(testStorage.getNamespaceName()).listPersistentVolumeClaims(testStorage.getNamespaceName(), testStorage.getClusterName()).stream().filter(
+            persistentVolumeClaim -> persistentVolumeClaim.getMetadata().getName().contains(testStorage.getClusterName())).collect(Collectors.toList());
 
-        if (!Environment.isKRaftModeEnabled()) {
-            RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getZookeeperSelector(), 1, zkPods);
+        for (PersistentVolumeClaim pvc : pvcs) {
+            LOGGER.info("Getting labels from {} pvc", pvc.getMetadata().getName());
+            verifyAppLabels(pvc.getMetadata().getLabels());
         }
-        RollingUpdateUtils.waitTillComponentHasRolled(testStorage.getNamespaceName(), testStorage.getKafkaSelector(), 3, kafkaPods);
-
-        labels.remove(labelKeys[0]);
-        labels.remove(labelKeys[1]);
-        labels.remove(labelKeys[2]);
 
-        LOGGER.info("Waiting for kafka service labels deletion {}", labels.toString());
-        ServiceUtils.waitForServiceLabelsDeletion(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()), labelKeys[0], labelKeys[1], labelKeys[2]);
+        LOGGER.info("---> Test Customer specified labels <--");
 
-        LOGGER.info("Verifying kafka labels via services");
-        service = kubeClient(testStorage.getNamespaceName()).getService(testStorage.getNamespaceName(), KafkaResources.brokersServiceName(testStorage.getClusterName()));
+        LOGGER.info("---> STRIMZI POD SETS <---");
 
-        verifyNullLabels(labelKeys, service);
+        LOGGER.info("Waiting for kafka Strimzi pod set labels existence {}", customSpecifiedLabels);
```

Comment:
```suggestion
        LOGGER.info("Waiting for Kafka StrimziPodSet labels existence {}", customSpecifiedLabels);
```

maybe add podset name and namespace?